### PR TITLE
Backport whitelist for SQL functions to ckan-2.7.12-opengov

### PR DESCRIPTION
**Pull request by peterVorman**
Fixes #

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Review by surajdeval — APPROVED** [review]
LGTM

File: ckanext/datastore/backend/postgres.py [hunk 1]
```diff
@@ -1555,8 +1555,17 @@ def search_sql(context, data_dict):
...
-        table_names, function_names = get_names(context, sql)
+        function_names = datastore_helpers.get_function_names_from_sql(context, sql)
+        for f in function_names:
+            for name_variant in [f.lower(), '"{}"'.format(f)]:
+                if name_variant in backend.allowed_sql_functions:
+                    break
+            else:
+                raise toolkit.NotAuthorized({
+                    'permissions': ['Not authorized to call function {}'.format(f)]
+                })
```
Comment (jguo144) [inline ckanext/datastore/backend/postgres.py]: I see this logic already used in the lines below. Does this need to be duplicated or should one be removed?
https://github.com/OpenGov-OpenData/ckan/blob/ff034c374376d7fecadaf74f36887ac4785334fa/ckanext/datastore/backend/postgres.py#L1579-L1587

File: ckanext/datastore/helpers.py [hunk 2]
```diff
@@ -67,6 +67,24 @@ def should_fts_index_field_type(field_type):
...
+                if function_name not in function_names:
+                    function_names.append(function_name)
+            if hasattr(token, 'tokens'):
+                return _get_function_names(token.tokens, function_names)
+
+        return function_names
+
+    parsed = sqlparse.parse(sql)[0]
+
+    return _get_function_names(parsed.tokens, [])
```
Comment (jguo144) [inline ckanext/datastore/helpers.py]: I'm seeing duplicated logic in the lines below
https://github.com/OpenGov-OpenData/ckan/blob/ff034c374376d7fecadaf74f36887ac4785334fa/ckanext/datastore/helpers.py#L151-L166

Comment (peterVorman) [reply ckanext/datastore/helpers.py]: removed

File: ckanext/datastore/backend/postgres.py (hunk 1 above)
Comment (peterVorman) [reply ckanext/datastore/backend/postgres.py]: removed

**Pull request merged**